tools/hidl/hidlcomp.cpp

```diff
@@ -3479,8 +3479,22 @@ void EspMessageInfo::write_esp()
     indentOuts(-1,"}\n");
         
     indentOuts("if (flags & 0x01) {\n");
+    bool isExtComplexType = true; //a complex type with no children, no attribute and no parent
+    if (isExtSimpleType || getParentName() || hasNonAttributeChild() || (espm_type_==espm_response && getMetaInt("exceptions_inline", 0)))
```

Comment:
The difference is "inherited parent" vs. "container parent".  I think we'll be fine.  If the rare case ever happens we can decide what to do then.